src/TesApi.Web/BatchScheduler.cs

```diff
@@ -34,14 +34,14 @@ public class BatchScheduler : IBatchScheduler
         private const string DownloadFilesScriptFileName = "download_files_script";
         private const string DockerInDockerImageName = "docker";
         private const string BlobxferImageName = "mcr.microsoft.com/blobxfer";
+        
         private static readonly Regex queryStringRegex = new Regex(@"[^\?.]*(\?.*)");
         private static readonly Regex externalStorageContainerRegex = new Regex(@"(https://([^\.]*).blob.core.windows.net/)([^\?/]+)/*?(\?.+)");
         private static readonly TimeSpan sasTokenDuration = TimeSpan.FromDays(3);
+        
         private readonly ILogger logger;
-        private readonly IAzureProxy azureProxy;
+        private readonly IAzureProxy cachingAzureProxy;
```

Comment:
I'd still call this azureProxy. BatchScheduler does not need to know if it is using the caching one or the regular one.